src/urllib3/util/util.py

```diff
@@ -0,0 +1,37 @@
+from typing import Optional, Union
+
+
+def to_bytes(
+    x: Union[str, bytes], encoding: Optional[str] = None, errors: Optional[str] = None
+) -> bytes:
+    if isinstance(x, bytes):
+        return x
+    elif not isinstance(x, str):
+        raise TypeError(f"not expecting type {type(x).__name__}")
+    elif encoding or errors:
+        return x.encode(encoding or "ascii", errors=errors or "strict")
+    return x.encode()
```

Comment:
Since we want to call `x.encode()`, because it's faster (based on @sethmlarson [comment](https://github.com/urllib3/urllib3/pull/2078#discussion_r527957964). I think we can change it like this:

`return x.encode(encoding or "utf-8", errors=errors or "strict")`

Reply:
Wow, I didn't realize the default encoding for `str.encode()` was utf-8, thought it was ascii all this time. Thanks for bringing this to my attention!

In this case we will need to explicitly say `to_str("ascii")` in all the places I previously thought should be ASCII.